java/android/hello/src/main/java/com/zeroc/hello/HelloApp.java

```diff
@@ -74,70 +70,55 @@ else if(m.what == MSG_EXCEPTION || m.what == MSG_RESPONSE)
             }
         };
 
-        _wifiManager = (WifiManager)getApplicationContext().getSystemService(Context.WIFI_SERVICE);
-
-        //
-        // On some devices, a multicast lock must be acquired otherwise multicast packets are discarded.
-        // The lock is initially unacquired.
-        //
-        _lock = _wifiManager.createMulticastLock("com.zeroc.demos.android.hello");
-        _lock.acquire();
-
         // SSL initialization can take some time. To avoid blocking the
         // calling thread, we perform the initialization in a separate thread.
-        new Thread(new Runnable()
-        {
-            public void run()
+        new Thread(() -> {
```

Comment:
We don't need to acquire the WifiManager lock ?

Reply:
Yes we do, good catch!!